### ISSUE TITLE
HASHED_PASSWORD environment variable does not work (v4.7.1)

### ISSUE THREAD

**Issue by bladerail** [linuxserver/docker-code-server#131]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

PASSWORD environment variable works, but HASHED_PASSWORD does not.
------------------------------

## Expected Behavior
1. I generate an argon2 hashed password.
2. I set the HASHED_PASSWORD environment variable in the docker-compose.yml.
3. I start the compose, access the web ui and key in the password.
4. The login succeeds and I get to use VSCode in the browser.

## Current Behavior
I am trying to run code-server on a Raspberry Pi 4B running 32-bit Raspbian. I generated a password by following the guide on this [link](https://github.com/coder/code-server/blob/main/docs/FAQ.md#can-i-store-my-password-hashed)
`echo -n "password" | npx argon2-cli -e`
Generated hash: `$argon2i$v=19$m=4096,t=3,p=1$sCgTPZTmOfDKWfBdpM+DBg$HnEDsCFdIUjLKfwiWHmx5/iRbPG8FM2aAvku5Ou8zas`. 
I verified the above hash by using the validator on [argon2 online](https://argon2.online/).

I can access the web gui showing the code-server prompting me to log in with a password, but get "Incorrect Password" no matter what I try.

## Steps to Reproduce

1. In the docker-compose.yml, set the HASHED_PASSWORD environment variable to `- HASHED_PASSWORD: "$argon2i$v=19$m=4096,t=3,p=1$sCgTPZTmOfDKWfBdpM+DBg$HnEDsCFdIUjLKfwiWHmx5/iRbPG8FM2aAvku5Ou8zas"`
2. Run the docker-compose. Docker logs indicate that HASHED_PASSWORD is being used. Access http://localhost:8443
5. Type "password" into the password field and Submit
6. Get "Incorrect Password" error.

I was able to get the PASSWORD environment variable working in plain text, but cannot get the HASHED_PASSWORD to work at all. I have tried deleting the config.yaml file, [having both PASSWORD and HASHED_PASSWORD environment variables on at the same time](https://github.com/coder/code-server/discussions/2666), tried using different hashes, and none of them have worked.

The config.yam file in /config/.config/code-server/config.yaml also doesn't seem to affect anything, is this correct behaviour? I have tried deleting the file then restarting the container, and also changing the `password` to hashed-password` in that file. The container seems to be using the password from somewhere else, because I can change the PASSWORD environment variable and that file never changes, but the login password actually does change. I'd like more clarification on whether the config.yaml file is used at all in this docker image.

## Environment
**OS:** Raspbian
**CPU architecture:** arm32
**How docker service was installed:**
Regular `get-docker.sh`

## Command used to create docker container (run/create/compose/screenshot)
`docker-compose up -d`

## Docker logs

```
      
[custom-init] No custom services found, skipping...
[migrations] started
[migrations] no migrations found
-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/
Brought to you by linuxserver.io
-------------------------------------
To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------
User uid:    1000
User gid:    1000
-------------------------------------
[custom-init] No custom files found, skipping...
[2022-10-15T08:42:17.247Z] info  code-server 4.7.1 77bbed48315a7cc275dc05a53d197197928f4b88
[2022-10-15T08:42:17.250Z] info  Using user-data-dir ~/data
[2022-10-15T08:42:17.296Z] info  Using config file ~/.config/code-server/config.yaml
[2022-10-15T08:42:17.297Z] info  HTTP server listening on http://0.0.0.0:8443/ 
[2022-10-15T08:42:17.297Z] info    - Authentication is enabled
[2022-10-15T08:42:17.297Z] info      - Using password from $HASHED_PASSWORD
[2022-10-15T08:42:17.297Z] info    - Not serving HTTPS 
[ls.io-init] done.
[16:50:26] Extension host agent started.
Failed login attempt {"remoteAddress":"10.0.0.2","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36","timestamp":1665823830}

      

```

**Comment by aptalca** [comment]
You need to escape the `$` character in compose yaml
https://docs.linuxserver.io/general/docker-compose#intro

**Comment by bladerail** [comment]
I forgot to mention in my original post that I had tried to escape the dollar sign several times yesterday, but none of them worked for me, but thanks to your suggestion, I went ahead and tried again. After several variations of escaping dollar signs (wasn’t sure if I needed the quotes, or to escape every single dollar), the final working version is

`- HASHED_PASSWORD=$$argon2i$$v=19$$m=4096,t=3,p=1$$sCgTPZTmOfDKWfBdpM+DBg$$HnEDsCFdIUjLKfwiWHmx5/iRbPG8FM2aAvku5Ou8zas`

Escape every single dollar, and have no quotes surrounding the environment variable.

My problem has been resolved, thank you very much.
